openedx/features/student_account/tests/test_set_opt_in_and_affiliate_user_organization.py

```diff
@@ -0,0 +1,135 @@
+        organization in the UserExtendedProfile and the EmailPreferences are
+        also set for that user
+        """
+
+        self.request_data['org_name'] = 'new_organization'
+        self.request_data['org_size'] = '5-10'
+        self.request_data['org_type'] = 'IWRNS'
+
+        request = RequestFactory().post('/user_api/v1/account/registration/', self.request_data)
+        params = dict(request.POST.copy().items())
+        params['name'] = '{} {}'.format(params.get('first_name'), params.get('last_name'))
+
+        form = AccountCreationFormCustom(data=params,
+                                         extended_profile_fields=None,
+                                         do_third_party_auth=False)
+        form.is_valid()
+
+        set_opt_in_and_affiliate_user_organization(self.user, form)
+        created_organization = Organization.objects.filter(label=form.cleaned_data.get('org_name')).first()
+
+        user_extended_profile_data = {
+            'is_first_learner': True,
+            'organization_id': created_organization.id
+        }
+
+        user_extended_profile_create_method.assert_called_once_with(user=self.user, **user_extended_profile_data)
+        email_preferences_create_method.assert_called_once_with(user=self.user,
+                                                                opt_in=form.cleaned_data.get('opt_in'))
+
+    @patch('openedx.features.student_account.helpers.EmailPreference.objects.create')
+    @patch('openedx.features.student_account.helpers.UserExtendedProfile.objects.create')
+    def test_user_with_existing_organization(self, user_extended_profile_create_method,
+                                             email_preferences_create_method):
+        """
+        Test that when a user gives the name of an already existing organization,
+        they are affiliated with that organization. The form does not raise any validation
+        errors, the EmailPreferences are also set for that user.
+        """
+
+        existing_organization = Organization.objects.create(label='existing_organization')
+        self.request_data['org_name'] = existing_organization.label
+
+        request = RequestFactory().post('/user_api/v1/account/registration/', self.request_data)
+        params = dict(request.POST.copy().items())
+        params['name'] = '{} {}'.format(params.get('first_name'), params.get('last_name'))
+
+        form = AccountCreationFormCustom(data=params,
+                                         extended_profile_fields=None,
+                                         do_third_party_auth=False)
+        form.is_valid()
+        set_opt_in_and_affiliate_user_organization(self.user, form)
+
+        user_extended_profile_data = {
+            'organization_id': existing_organization.id
+        }
+
+        user_extended_profile_create_method.assert_called_once_with(user=self.user, **user_extended_profile_data)
```

Comment:
Let's cater all scenarios. Test for following line of code is missing for existing organization

```
    if org_size:
        user_organization.total_employees = org_size

    if org_type:
        user_organization.org_type = org_type